src/Infrastructure/BotSharp.MCP/AIFunctionUtilities.cs

```diff
@@ -0,0 +1,37 @@
+using BotSharp.Abstraction.Functions.Models;
+using McpDotNet.Protocol.Types;
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace BotSharp.Core.MCP;
+
+internal static class AIFunctionUtilities
+{
+    public static FunctionDef MapToFunctionDef(Tool tool)
+    {
+        if (tool == null)
+        {
+            throw new ArgumentNullException(nameof(tool));
+        }
+
+        var properties = tool.InputSchema?.Properties;
+        var required = tool.InputSchema?.Required ?? new List<string>();
```

Comment:
Is the list readonly? If yes, just use singleton Array.Empty<string>() here.

Reply:
list not readonly